From c729ee24a6d6e49b1d7378c8662ed86e0961e737 Mon Sep 17 00:00:00 2001
From: Sebastian Reichel <sebastian.reichel@collabora.com>
Date: Mon, 24 Aug 2026 19:07:24 +0200
Subject: [PATCH 131/137] arm64: dts: rockchip: add missing codec supplies for
 Rock 5B family

Describe the es8316 codec supplies to fix the following kernel warning:

es8316 7-0011: supply dvdd not found, using dummy regulator
es8316 7-0011: supply pvdd not found, using dummy regulator

Regulator information was taken from schematics of 5B, 5B+ and 5T.

Signed-off-by: Sebastian Reichel <sebastian.reichel@collabora.com>
---
 .../boot/dts/rockchip/rk3588-rock-5b-5bp-5t.dtsi | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b-5bp-5t.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b-5bp-5t.dtsi
index 8635b45..7dacd49 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b-5bp-5t.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b-5bp-5t.dtsi
@@ -15,6 +15,14 @@ aliases {
 		mmc2 = &sdio;
 	};
 
+	avcc_1v8_codec_s0: avcc-1v8-codec-s0 {
+		compatible = "regulator-fixed";
+		regulator-name = "avcc_1v8_codec_s0";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		vin-supply = <&vcc_1v8_s0>;
+	};
+
 	chosen {
 		stdout-path = "serial2:1500000n8";
 	};
@@ -422,10 +430,14 @@ &i2c7 {
 	es8316: audio-codec@11 {
 		compatible = "everest,es8316";
 		reg = <0x11>;
-		clocks = <&cru I2S0_8CH_MCLKOUT>;
-		clock-names = "mclk";
 		assigned-clocks = <&cru I2S0_8CH_MCLKOUT>;
 		assigned-clock-rates = <12288000>;
+		avdd-supply = <&vcc_3v3_s0>;
+		clocks = <&cru I2S0_8CH_MCLKOUT>;
+		clock-names = "mclk";
+		cpvdd-supply = <&avcc_1v8_codec_s0>;
+		dvdd-supply = <&avcc_1v8_codec_s0>;
+		pvdd-supply = <&avcc_1v8_codec_s0>; // via vcc_1v8_s0
 		#sound-dai-cells = <0>;
 
 		port {
-- 
2.47.3

